app/models/product_import/entry_validator.rb

```diff
@@ -67,10 +67,15 @@ def enterprise_field
     end
 
     def mark_as_new_variant(entry, product_id)
+      # Variant needs a product. Product needs to be assigned first in order for
+      # delegate to work. name= will fail otherwise.
       new_variant = Spree::Variant.new(
-        entry.assignable_attributes.except('id', 'product_id', 'on_hand', 'on_demand',
-                                           'variant_unit', 'variant_unit_name',
-                                           'variant_unit_scale', 'primary_taxon_id')
+        { product_id: }
+        .merge(
+          entry.assignable_attributes.except('id', 'product_id', 'on_hand', 'on_demand',
+                                             'variant_unit', 'variant_unit_name',
+                                             'variant_unit_scale', 'primary_taxon_id')
+        )
```

Comment:
I think another way to express this is like this with the "double-splat" operator. Also we might as well load the desired attributes in a separate step. Ok, I couldn't help but give it a try, it's cathartic. Hope you don't mind :)

Reply:
Sorry, I just noticed you're working on this at the same time, I'll step away and check later 😉

Reply:
No, should be ok. I just rebased it. Give it a shot.

Reply:
Thanks, done, let me know if the added commit looks ok to you.

Reply:
Looks like a big improvement. It looks much nicer.